onadata/apps/api/tests/viewsets/test_user_profile_viewset.py

```diff
@@ -1572,3 +1571,79 @@ def test_password_optional(self, mock_send_verification_email):
         self.assertTrue(mock_send_verification_email.called)
         user = User.objects.get(username="deno")
         self.assertTrue(user.is_active)
+
+    def test_cache_isolation_between_users_retrieve(self):
+        """Test that user A's cached profile is not accessible to user B"""
+        # Create user A
+        user_a_data = _profile_data()
+        user_a_data["username"] = "user_a"
+        user_a_data["email"] = "user_a@example.com"
+        user_a_data["first_name"] = "UserA"
+        user_a_data["name"] = "UserA lastname"
+        self._create_user_using_profiles_endpoint(user_a_data)
+        user_a = User.objects.get(username="user_a")
+        token_a = Token.objects.get(user=user_a)
+        user_a_extra = {"HTTP_AUTHORIZATION": f"Token {token_a.key}"}
+
+        # Create user B
+        user_b_data = _profile_data()
+        user_b_data["username"] = "user_b"
+        user_b_data["email"] = "user_b@example.com"
+        user_b_data["first_name"] = "UserB"
+        user_b_data["name"] = "UserB lastname"
+        self._create_user_using_profiles_endpoint(user_b_data)
+        user_b = User.objects.get(username="user_b")
+        token_b = Token.objects.get(user=user_b)
+        user_b_extra = {"HTTP_AUTHORIZATION": f"Token {token_b.key}"}
+
+        cache.clear()
+        retrieve_view = UserProfileViewSet.as_view({"get": "retrieve"})
+
+        # User A updates their profile and caches it
```

Comment:
Can we update the cache directly here without making a call to `PATCH`? We are only interested in cache data which means we can work with `cache.set` directly to simulate cached data?